### ISSUE TITLE
0.8.6 depends_on not working for aws_appautoscaling_policy and aws_appautoscaling_target

### ISSUE THREAD

**Issue by hashibot** [terraform-providers/terraform-provider-aws#538]
_This issue was originally opened by @eedwardsdisco as hashicorp/terraform#11987. It was migrated here as part of the [provider split](https://www.hashicorp.com/blog/upcoming-provider-changes-in-terraform-0-10/). The original body of the issue is below._

<hr>

assuming I have the two following resources (no other attributes, for clarity):

```
resource "aws_appautoscaling_policy" "scale_in_web" {
	depends_on = ["aws_appautoscaling_target.web"]
}

```

```
resource "aws_appautoscaling_target" "web" {
}
```

case:
when terraform sees a change in the aws_appautoscaling_target and it needs to re-create it...

expected behavior:
terraform respects the explicit dependency, and will re-create the aws_appautoscaling_policy.web after re-creating the aws_appautoscaling_target.web

actual behavior:
terraform ignores the explicit dependency, and will re-create the aws_appautoscaling_target.web and leave the aws_appautoscaling_policy.web in a broken state

I was able to work around this by manually tainting the policy, but this is the behavior I expect to happen automatically, based on the explicit dependency.

**Comment by xavivars** [comment]
@eedwardsdisco how did you workaround on this? I didn't manage to make the aws_appautoscaling_policy really depend on the aws_appautoscaling_target

**Comment by bflad** [comment]
Hi folks 👋 Sorry for the unexpected behavior here. It turns out there was a very subtle bug that prevented proper recreation of Application AutoScaling Policies when only the `resource_id` argument changed. We also now properly ignore `ObjectNotFoundException` errors during Terraform resource destroy. A full writeup of these changes can be found in #7982. These fixes will be released in version 2.3.0 of the Terraform AWS Provider in the next day or two.

Even without upgrading your Terraform AWS Provider to the newer version, you _may_ be able to workaround the original issue with a simple configuration update. Using direct references to the relevant `aws_appautoscaling_target` resource for the `resource_id`, `scalable_dimension`, and `service_namespace` arguments in the `aws_appautoscaling_policy` resource configuration should be enough in many cases to ensure Terraform has proper ordering information to handle recreations of the Application AutoScaling Policy resource when the underlying Application AutoScaling Target or resource associated with the target changes:

```hcl
resource "aws_appautoscaling_target" "example" {
  # ... other configuration ...
}

resource "aws_appautoscaling_policy" "example" {
 # ... other configuration ...

  resource_id        = "${aws_appautoscaling_target.example.resource_id}"
  scalable_dimension = "${aws_appautoscaling_target.example.scalable_dimension}"
  service_namespace  = "${aws_appautoscaling_target.example.service_namespace}"
}
```

If you are still having trouble after upgrading to version 2.3.0 of the Terraform AWS Provider (when its released) and with a configuration looking similar to the above, please create a new GitHub issue with the relevant details from the issue template and we can further triage. Thanks!

**Comment by bflad** [comment]
This has been released in [version 2.3.0 of the Terraform AWS provider](https://github.com/terraform-providers/terraform-provider-aws/blob/master/CHANGELOG.md#230-march-21-2019). Please see the [Terraform documentation on provider versioning](https://www.terraform.io/docs/configuration/providers.html#provider-versions) or reach out if you need any assistance upgrading.
